sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py

```diff
@@ -199,6 +199,8 @@ def _wrap_eventdata(
             if isinstance(
                 event_data, EventDataBatch
             ):  # The partition_key in the param will be omitted.
+                if not event_data:
```

Comment:
Does this work? From what I can see `EventDataBatch` has no `__bool__`?
Or will it fallback to checking `__len__`?

Reply:
the latter one, it fallbacks to `__len__`